services/http/api/lib/models/initializers.ts

```diff
@@ -0,0 +1,202 @@
+            label: 'Measurement',
+            schema: {
+                type: {
+                    type: 'string',
+                    required: true
+                },
+                value: {
+                    type: 'number',
+                    required: true
+                }
+            },
+            relationships: {
+                Date: {
+                    model: dateModel,
+                    direction: 'out',
+                    name: 'RecordedAt',
+                    properties: {
+                        time: {
+                            property: 'time',
+                            schema: {
+                                type: 'number'
+                            }
+                        }
+                    }
+                }
+            },
+            primaryKeyField: 'type'
+        },
+        db
+    );
+}
+
+export function initResourceModel(db: Neogma): resource.ResourceModel {
+    return ModelFactory<resource.ResourceProperties, resource.ResourceRelatedNode>(
+        {
+            label: 'Resource',
+            schema: {
+                path: {
+                    type: 'string',
+                    required: true
+                }
+            },
+        },
+        db
+    )
+}
+
+export function initUserModel(db: Neogma, measurementModel: measurement.MeasurementModel, resourceModel: resource.ResourceModel): user.UserModel {
+    return ModelFactory<user.UserProperties, user.UserRelatedNode>(
+        {
+            label: 'User',
+            schema: {
+                uid: {
+                    type: 'string',
+                    required: true,
+                },
+            },
+            relationships: {
+                Measurement: {
+                    model: measurementModel,
+                    direction: 'out',
+                    name: 'Recorded',
+                },
+                Resource: {
+                    model: resourceModel,
+                    direction: 'out',
+                    name: 'Can',
+                    properties: {
+                        method: {
+                            property: 'method',
+                            schema: {
+                                type: 'string'
+                            }
+                        }
+                    }
+                }
+            },
+            primaryKeyField: 'uid',
+            // relationshipCreationKeys: {}
+        },
+        db
+);
+}
+
+
+export type allModels = {
+    date: date.DateModel,
+    identity: identity.IdentityModel,
+    measurement: measurement.MeasurementModel,
+    resource: resource.ResourceModel,
+    user: user.UserModel,
+}
+export function initAllModels(db: Neogma): allModels {
+    const date = initDateModel(db);
+    
+    const resource = initResourceModel(db);
+    const measurement = initMeasurementModel(db, date);
+    const identity = initIdentityModel(db, resource);
+    const user = initUserModel(db, measurement, resource);
```

Comment:
It's not as simple as that unfortunately, some of these models need an input of other initialized models to create relationships with each other. (That's why some initializer functions come after other initialized models). This is vital because it gives neogma a reference to how the relationship should be made. 

I've been defining outgoing relationships by default (because it's how I mainly look at them). I'm not sure if it's necessary though.

It's also important to define them if you are creating a node, but it requires an incoming relation from some other node.
For example, 
`[Source] -> [MeasurementType] -> [Measurement] <- [User]`
If you are creating measurement, then merging type, source and user to that new node, measurement needs to know that it has an incoming relation from type and user. MeasurementType also needs to have an incoming relation from source.

Reply:
I think it could be kept the way it is right now (and add documentation on how to expand on this), or I could remove all relationship building in the initializing stage, and just add them all afterwards which simplifies the that stage, but makes the add--more-relationship part messy.

Reply:
If we didn't do it in the initialization where would it happen? Sounds like the best course of action is just to create a steps document though.

Reply:
It would still happen at the initialization stage, except since there is 2 parts (initializing them individually, and bundling that all up in the `initAllModels` function, it would just happen outside of the individual one. I've added a readme section for the time being though.

Reply:
Great